src/Build/BackEnd/Components/RequestBuilder/TargetUpToDateChecker.cs

```diff
@@ -351,8 +352,31 @@ private static string GetIncrementalBuildReason(DependencyAnalysisLogDetail logD
         /// </summary>
         private void LogUniqueInputsAndOutputs()
         {
-            _loggingService.LogComment(_buildEventContext, MessageImportance.Low, "SkipTargetUpToDateInputs", string.Join(";", _uniqueTargetInputs.Keys));
-            _loggingService.LogComment(_buildEventContext, MessageImportance.Low, "SkipTargetUpToDateOutputs", string.Join(";", _uniqueTargetOutputs.Keys));
+            var targetInputKeys = _uniqueTargetInputs.Keys;
+            var targetOutputKeys = _uniqueTargetOutputs.Keys;
+
+            var maxLengthSum = Math.Max(LengthSum(targetInputKeys), LengthSum(targetOutputKeys));
+            var maxLength = Math.Max(targetInputKeys.Count, targetOutputKeys.Count);
```

Comment:
name this `joinerCount` or `semicolonCount` or something? I thought the length calculation looked fishy because the name confused me.